universal-application-tool-0.0.1/app/forms/MultiOptionQuestionForm.java

```diff
@@ -143,27 +146,42 @@ public QuestionDefinitionBuilder getBuilder() {
       predicateBuilder.setMaxChoicesAllowed(getMaxChoicesAllowed());
     }
 
-    ImmutableList.Builder<QuestionOption> questionOptions = ImmutableList.builder();
+    ImmutableList.Builder<QuestionOption> questionOptionsBuilder = ImmutableList.builder();
     Preconditions.checkState(
         this.optionIds.size() == this.options.size(),
         "Option ids and options are not the same size.");
 
+    if (this.optionIds.size() > 0) {
+      setNextAvailableId(this.optionIds.get(0) + 1);
+    }
+
     // Note: the question edit form only sets or updates the default locale.
     for (int i = 0; i < options.size(); i++) {
-      questionOptions.add(
+      questionOptionsBuilder.add(
           QuestionOption.create(
               optionIds.get(i), i, LocalizedStrings.withDefaultValue(options.get(i))));
     }
+
+    // The IDs are not guaranteed to be in any type of order, so doing this ensures that we find
+    // the largest ID in the list and accurately set the next largest.
+    Long maxId = optionIds.stream().max(Long::compareTo).orElse(0L);
+    setNextAvailableId(maxId + 1);
+
     for (int i = 0; i < newOptions.size(); i++) {
-      questionOptions.add(
+      questionOptionsBuilder.add(
           QuestionOption.create(
               nextAvailableId.orElse(0L) + i,
```

Comment:
This orElse isn't necessary anymore, and if it's actually used it'd indicate a coding error since we just set it above, so we'd prefer the exception rather than the incorrect data.

Reply:
Done!